Classes/Widgets/ContentChangesWidget.php

```diff
@@ -0,0 +1,142 @@
+        $query = $queryBuilder->select(...$labelFieldsForQuery)
+            ->from($table, 't')
+            ->orWhere(
+                $queryBuilder->expr()->gte('t.starttime', $queryBuilder->createNamedParameter($now, \PDO::PARAM_INT)),
+                $queryBuilder->expr()->gte('t.endtime', $queryBuilder->createNamedParameter($now, \PDO::PARAM_INT))
+            )->setMaxResults($this->limit);
+
+        return $this->processItemsQuery($table, $query);
+    }
+
+    protected function getMostRecentChangedRecordsFromTable(string $table): array
+    {
+        $labelFieldsForQuery = $this->resolveLabelFieldsForTable($table);
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable($table);
+        $query = $queryBuilder->select('u.username', ...$labelFieldsForQuery)
+            ->from($table, 't')
+            ->rightJoin(
+                't',
+                'sys_log',
+                'l',
+                (string) $queryBuilder->expr()->andX(
+                    $queryBuilder->expr()->eq('l.recuid', 't.uid'),
+                    $queryBuilder->expr()->eq('l.tablename', $queryBuilder->createNamedParameter($table, \PDO::PARAM_STR))
+                )
+            )
+            ->leftJoin(
+                'l',
+                'be_users',
+                'u',
+                'u.uid = l.userid'
+            )
+            ->orderBy('t.tstamp', 'DESC')
+            ->groupBy('t.uid', 't.pid', 't.sys_language_uid', 'u.username', ...$labelFieldsForQuery)
+            ->setMaxResults($this->limit);
+
+        return $this->processItemsQuery($table, $query);
+    }
+
+    protected function resolveLabelFieldsForTable(string $table): array
+    {
+        $labelFields = [$GLOBALS['TCA'][$table]['ctrl']['label']] + explode(',', $GLOBALS['TCA'][$table]['ctrl']['label_alt'] ?? '');
+        $labelFields = array_filter($labelFields);
+        $labelFieldsForQuery = [];
+        foreach ($labelFields as $labelField) {
+            $labelFieldsForQuery[] = 't.' . $labelField;
+        }
+        return $labelFieldsForQuery;
+    }
+
+    protected function processItemsQuery(string $table, QueryBuilder $query): array
+    {
+        $iconFactory = $this->getIconFactory();
+
+        $query->addSelect('t.uid', 't.pid', 't.tstamp', 't.sys_language_uid', 't.starttime', 't.endtime');
+        if ($table === 'pages') {
+            $query->addSelect('t.doktype', 't.is_siteroot');
+        } elseif ($table === 'tt_content') {
+            $query->addSelect('t.CType');
+        }
+
+        $results = $query->execute()->fetchAll();
+        if (!$results) {
+            return [];
+        }
+        foreach ($results as &$result) {
+            $result['type'] = $table;
+            $result['label'] = BackendUtility::getRecordTitle($table, $result);
+            $result['icon'] = $iconFactory->getIconForRecord($table, $result, Icon::SIZE_SMALL);
```

Comment:
Not quite sure if we want coloured icons in the dashboard. We want to keep the dashboard simple and no overdue of color. On the other hand it might be a recognisable icon for the user.

@irenesacchi what do you think?